abs/replica_client.go

```diff
@@ -387,6 +407,21 @@ func (itr *ltxFileIterator) loadNextPage() bool {
 			continue
 		}
 
+		// Use fast timestamp from blob creation by default
+		createdAt := item.Properties.CreationTime.UTC()
+
+		// Only read accurate timestamp from metadata when requested (timestamp-based restore)
+		// Azure includes metadata in LIST operations, so no extra API call needed
+		if !itr.timestamp.IsZero() && item.Metadata != nil {
+			if ts, ok := item.Metadata[litestream.MetadataKeyTimestampAzure]; ok && ts != nil {
+				if parsed, err := time.Parse(time.RFC3339Nano, *ts); err == nil {
+					createdAt = parsed
+				}
+			}
+		}
```

Comment:
I don't think it's worth having `c.timestamp` in this iterator since it's zero cost to grab the correct timestamp from the metadata. 

```go
info.CreatedAt = item.Properties.CreationTime.UTC()
if item.Metadata != nil {
	if ts, ok := item.Metadata[litestream.MetadataKeyTimestampAzure]; ok && ts != nil {
		if parsed, err := time.Parse(time.RFC3339Nano, *ts); err == nil {
			info.CreatedAt = parsed
		}
	}
}
```